core/backend/src/CheckpointManager.ts

```diff
@@ -149,30 +149,14 @@ export class V2CheckpointManager {
     this.containers.clear();
   }
 
-  public static getFileName(checkpoint: CheckpointProps): LocalFileName {
-    const changesetId = checkpoint.changeset.id || "first";
-    return path.join(this.getFolder(), `${changesetId}.bim`);
-  }
-
   private static get cloudCache(): CloudSqlite.CloudCache {
     if (!this._cloudCache) {
-      let cacheDir = process.env.CHECKPOINT_CACHE_DIR;
-      if (!cacheDir) {
-        cacheDir = this.getFolder();
-        Logger.logWarning(loggerCategory, `No CHECKPOINT_CACHE_DIR found in process.env, using ${cacheDir} instead.`);
-      } else {
-        // Make sure the checkpoint_cache_dir has an iTwinDaemon specific file in it, otherwise fall back to other directory.
-        if (!(IModelJsFs.existsSync(path.join(cacheDir, "portnumber.bcv")))) {
-          cacheDir = this.getFolder();
-          Logger.logWarning(loggerCategory, `No evidence of the iTwinDaemon in provided CHECKPOINT_CACHE_DIR: ${process.env.CHECKPOINT_CACHE_DIR}, using ${cacheDir} instead.`);
-        }
-      }
+      let cacheDir: string | undefined = process.env.CHECKPOINT_CACHE_DIR ?? this.getFolder();
+      // See if there is a daemon running, otherwise use profile directory for cloudCache
+      if (!(IModelJsFs.existsSync(path.join(cacheDir, "portnumber.bcv"))))
```

Comment:
> for the daemonless case and why you want it to live under a profile directory

yes, that's why profiles exist. No two processes may share a cloud cache. 

> Would we be better off just encouraging downloading checkpoints to a shared directory if they're only going to be opened readonly anyways?

No, you're *way* better off opening a checkpoints from their container rather than downloading them, because: 
 - startup is faster. Often people will only need a few blocks. Prefetching works if you need more.
 - disk space management. This is nearly impossible without cloud caches.
 - blocks are shared across checkpoints. This will be the biggest benefit for many desktop use cases where people want to compare versions.
 
In the (rare) case where people use the same checkpoint with multiple profiles, it will use a little more disk space. That is not an issue.

Reply:
It turns out that it is not really possible for two processes to share the same downloaded checkpoint at the same time anyway, even if they both open it for read. That's because we create and write to "cache" files (e.g. ".Tiles") during the session. I'm not sure how we can stop that problem